.github/workflows/build-and-deploy.yml

```
@@ -2,16 +2,79 @@ name: Build & Deploy
 
 on:
   push:
+    branches:
+      - main
+      - develop
+    paths:
+      - 'client/**'
+      - 'server/**'
+      - 'shared/**'
+      - '.github/workflows/**'
     branches: [main, develop]
```

actionlint (1.7.10)

[error] key "branches" is duplicated in "push" section. previously defined at line:5,col:5

(syntax-check)